orca-so/whirlpools
# Add Pool Test for Rust-sdk

# Add pool test utilities and test cases

## Overview
Add pool initialization test utilities and corresponding test cases to validate core pool functionality, mirroring the TypeScript SDK's test coverage.

## Core changes
Implement setup_whirlpool utility function that:
- Creates token vaults and initializes pool with specified token pair and tick spacing
- Handles both standard and concentrated liquidity pool configurations
- Added mock implementation of `getProgramAccounts` to handle both extensions and normal PDA

Add test cases to verify:

- Splash pool initialization and configuration
- Concentrated liquidity pool initialization and configuration
- Pool state validation including liquidity, tick spacing, and fee rates

Implement test_fetch_all_pools_for_pair test case:

- Add mock implementation of getProgramAccounts RPC method with filter support
- Enable testing of pool discovery functionality previously unavailable in ts-sdk
- Verify correct handling of both initialized and uninitialized pools

Fix bug in pool discovery:

- Update pool fetching logic to properly iterate through fee tiers
- Ensure uninitialized pools are correctly identified and returned
